runtime/src/accounts_file.rs

```diff
@@ -0,0 +1,179 @@
+use {
+    crate::{append_vec::*, storable_accounts::StorableAccounts},
+    solana_sdk::{account::ReadableAccount, clock::Slot, hash::Hash, pubkey::Pubkey},
+    std::{
+        borrow::Borrow,
+        io,
+        path::{Path, PathBuf},
+    },
+};
+
+#[derive(Debug)]
+/// An enum for accessing an accounts file which can be implemented
+/// under different formats.
+pub enum AccountsFile {
+    AppendVec(AppendVec),
+}
+
+impl AccountsFile {
+    pub fn new_append_vec(file: &Path, create: bool, size: usize) -> Self {
+        AccountsFile::AppendVec(AppendVec::new(file, create, size))
+    }
+
+    pub fn new_append_vec_from_file<P: AsRef<Path>>(
+        path: P,
+        current_len: usize,
+    ) -> io::Result<(Self, usize)> {
+        let (append_vec, len) = AppendVec::new_from_file(path, current_len)?;
+        Ok((AccountsFile::AppendVec(append_vec), len))
+    }
```

Comment:
I see your point.  The motivation here is to hide the details so that the call sites don't need to import `append::AppendVec` or other storage formats.

Reply:
That makes sense as well.  Will do!